src/lib/zxing/index.ts

```diff
@@ -0,0 +1,76 @@
+import type { BrowserQRCodeReader, IScannerControls } from '@zxing/browser'
+import type { Exception, Result } from '@zxing/library'
+
+type DecodeContinuouslyCallback = (
+  result?: Result,
+  error?: Exception,
+  controls?: IScannerControls
+) => void
+export class Scanner {
+  cameraStream!: MediaStream
+  codeReader!: BrowserQRCodeReader
+  videoElement: HTMLVideoElement
+
+  constructor({ videoElement }: { videoElement: HTMLVideoElement }) {
+    this.videoElement = videoElement
+  }
+
+  async init() {
+    const { BrowserQRCodeReader } = await import('@zxing/browser')
+    this.codeReader = new BrowserQRCodeReader()
+  }
+
+  async start(currentCamera: number | null, decodeCallback: DecodeContinuouslyCallback) {
+    // Stop all tracks from media stream before camera changing
+    if (this.cameraStream) this.stopVideoTracks()
+
+    // Request new media stream and attach to video element as source object
+    const facingMode = currentCamera === 1 ? 'environment' : 'user'
+    this.cameraStream = await navigator.mediaDevices.getUserMedia({
+      video: { facingMode },
+      audio: false
+    })
+    this.videoElement.srcObject = this.cameraStream
+
+    return this.codeReader.decodeFromVideoElement(this.videoElement, decodeCallback)
+  }
+
+  async getCameras() {
+    const deviceCameras = (await navigator.mediaDevices.enumerateDevices()).filter(
+      (device) => device.kind === 'videoinput'
+    )
+
+    // Change only two video devices. First - front camera, second (is available) - back camera
+    const cameras: MediaDeviceInfo[] = []
+    if (deviceCameras.length > 0) {
+      cameras.push(deviceCameras[0])
+
+      if (deviceCameras.length > 1) {
+        let backCamera
+        if (navigator.userAgent.indexOf('iPhone'.toLowerCase())) {
```

Comment:
`indexOf` may return `-1` when no element is found which leads to a truthy value. Use `includes` instead. It does return boolean.

Reply:
yes, this is typo